packages/gatsby/src/bootstrap/index.js

```diff
@@ -452,12 +452,21 @@ module.exports = async (args: BootstrapArgs) => {
     require(`./page-hot-reloader`)(graphqlRunner)
   }
 
-  // Run queries
-  activity = report.activityTimer(`run graphql queries`, {
+  const queryIds = pageQueryRunner.calcInitialDirtyQueryIds(store.getState())
+  const { staticQueryIds, pageQueryIds } = pageQueryRunner.groupQueryIds(
+    queryIds
+  )
+
+  activity = report.activityTimer(`run static queries`, {
     parentSpan: bootstrapSpan,
   })
   activity.start()
-  await runInitialQueries(activity)
+  await pageQueryRunner.processStaticQueries(staticQueryIds, { activity })
```

Comment:
What's the reason for not passing store.getState() in here?

```suggestion
  await pageQueryRunner.processStaticQueries(staticQueryIds, { activity, state: store.getState() })
```

Reply:
Mostly laziness :)